tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationResult.cs

```diff
@@ -5,27 +5,52 @@
 
 #nullable enable
 
+using System.Diagnostics.CodeAnalysis;
+
 namespace Datadog.Trace.Configuration.ConfigurationSources.Telemetry;
 
 internal readonly record struct ConfigurationResult<T>
 {
-    private ConfigurationResult(T result, bool isValid)
+    private ConfigurationResult(T? result, ConfigurationLoadResult loadResult)
     {
         Result = result;
-        IsValid = isValid;
+        LoadResult = loadResult;
     }
 
     /// <summary>
-    /// Gets the extracted configuration value.
+    /// Gets the extracted configuration value, if it was found
     /// </summary>
-    public T Result { get; }
+    public T? Result { get; }
 
     /// <summary>
     /// Gets a value indicating whether <see cref="Result"/> result passed validation.
+    /// If <c>true</c>, implies that <see cref="Result"/> contains a valid value. If
+    /// <c>false</c>, see <see cref="LoadResult"/> for more information.
+    /// </summary>
+    [MemberNotNullWhen(true, nameof(Result))]
+    public bool IsValid => LoadResult == ConfigurationLoadResult.Valid;
+
+    /// <summary>
+    /// Gets a value indicating whether the configuration loader should defer to a fallback key
+    /// </summary>
+    public bool ShouldFallBack => LoadResult is not (ConfigurationLoadResult.Valid or ConfigurationLoadResult.ValidationFailure);
```

Comment:
Can the check be re-written to this? I think it would be cleaner:
```suggestion
    public bool ShouldFallBack => LoadResult is ConfigurationLoadResult.NotFound or ConfigurationLoadResult.ParsingError;
```

Reply:
It can, and that's how I wrote it originally - I only changed it so that "if we add another error type the condition doesn't need to change". Which is a bad excuse for the worse readability, so will change it, thanks 👍 😄